src/main/java/com/dicemc/corruptedlands/blocks/mystical_pumpkins/CorruptedPumpkinBlock.java

```diff
@@ -0,0 +1,44 @@
+package com.dicemc.corruptedlands.blocks.mystical_pumpkins;
+
+import blueduck.mysticalpumpkins.block.MysticalPumpkinBlock;
+import com.dicemc.corruptedlands.Config;
+import com.dicemc.corruptedlands.CorruptedLandMod;
+import com.dicemc.corruptedlands.Registration;
+import com.dicemc.corruptedlands.blocks.ICorrupted;
+import com.dicemc.corruptedlands.blocks.IPreventCorruption;
+import net.minecraft.block.BlockState;
+import net.minecraft.enchantment.EnchantmentHelper;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.util.DamageSource;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import net.minecraft.world.server.ServerWorld;
+
+import java.util.Random;
+
+public class CorruptedPumpkinBlock extends MysticalPumpkinBlock implements ICorrupted, IPreventCorruption {
+	@Override
+	public void randomTick(BlockState state, ServerWorld worldIn, BlockPos pos, Random random) {
+		double threshold = random.nextDouble();
+		if (threshold >= Config.SPREAD_RATE.get()) CorruptedLandMod.Core.corruptNeighbors(pos, worldIn);
+	}
+	
+	@Override
+	public void onEntityWalk(World worldIn, BlockPos pos, Entity entityIn) {
```

Comment:
this is unnecessary if unless pumpkins should cause damage like magma blocks do.

Reply:
this check is for all blocks, and you don't have any corrupted magma as well as me not knowing what that specific method did
remove it if ya want ig